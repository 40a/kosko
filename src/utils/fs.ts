import fs from "fs";
import mkdirp from "mkdirp";
import { promisify } from "util";

export const access = promisify(fs.access);
export const readDir = promisify(fs.readdir);
export const stat = promisify(fs.stat);
export const readFile = promisify(fs.readFile);
export const writeFile = promisify(fs.writeFile);
export const mkdirs = promisify(mkdirp);

export async function exists(path: string, mode?: number) {
  try {
    await access(path, mode);
    return true;
  } catch (err) {
    if (err.code !== "ENOENT") throw err;
    return false;
  }
}
