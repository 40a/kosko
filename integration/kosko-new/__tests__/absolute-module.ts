import execa from "execa";
import { join } from "path";
import tmp from "tmp-promise";
import { runCLI } from "../../run";
import symlinkDir from "symlink-dir";
import pkgDir from "pkg-dir";
import fs from "fs";
import { promisify } from "util";

const readFile = promisify(fs.readFile);

let result: execa.ExecaReturns;
let tmpDir: tmp.DirectoryResult;

beforeAll(async () => {
  const root = await pkgDir();
  tmpDir = await tmp.dir({ unsafeCleanup: true });
  await symlinkDir(
    join(root!, "packages/template-environment"),
    join(tmpDir.path, "node_modules", "@kosko", "template-environment")
  );
  result = await runCLI(
    ["new", "@kosko/template-environment", "--name", "dev"],
    { cwd: tmpDir.path }
  );
});

afterAll(() => tmpDir.cleanup());

test("should return status code 0", () => {
  expect(result.code).toEqual(0);
});

test("should create files based on templates", async () => {
  const content = await readFile(
    join(tmpDir.path, "environments", "dev", "index.js"),
    "utf8"
  );
  expect(content).toMatchSnapshot();
});
