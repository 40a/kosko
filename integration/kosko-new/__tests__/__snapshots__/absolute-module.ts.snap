// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create files based on templates 1`] = `
"\\"use strict\\";

const requireDir = require(\\"require-dir\\");

exports.globals = {};

exports.components = requireDir(\\".\\", {
  mapValue: function(value) {
    return {
      ...exports.globals,
      ...(value.default || value)
    };
  }
});
"
`;
